src/prefect/runner/submit.py

```diff
@@ -0,0 +1,196 @@
+
+
+@sync_compatible
+async def submit_to_runner(
+    prefect_callable: Union[Flow, Task],
+    parameters: Dict[str, Any],
+) -> uuid.UUID:
+    """
+    Run a callable in the background via the runner webserver.
+
+    Args:
+        prefect_callable: the callable to run (only flows are supported for now, but eventually tasks)
+        parameters: the keyword arguments to pass to the callable,
+    """
+
+    flow_run_id = await _submit_flow_to_runner(prefect_callable, parameters)
+
+    if inspect.isawaitable(flow_run_id):
+        return await flow_run_id
+    else:
+        return flow_run_id
+
+
+@sync_compatible
+async def submit_many_to_runner(
+    prefect_callable: Union[Flow, Task], parameters_list: List[Dict[str, Any]]
+) -> List[uuid.UUID]:
+    """
+    Run multiple callables in the background via the runner webserver, respecting the run limit.
+
+    Args:
+        prefect_callable: the callable to run (flows or tasks)
+        parameters_list: a list of dictionaries, each representing keyword arguments to pass to the callable.
+    """
+
+    n_current_runs = await get_current_run_count()
+    available_slots = max(0, PREFECT_RUNNER_PROCESS_LIMIT.value() - n_current_runs)
+
+    submitted_run_ids = []
+    for parameters in parameters_list[:available_slots]:
+        flow_run_id = await submit_to_runner(prefect_callable, parameters)
+        if inspect.isawaitable(flow_run_id):
+            flow_run_id = await flow_run_id
+        submitted_run_ids.append(flow_run_id)
+
+    if (diff := len(parameters_list) - available_slots) > 0:
+        logger.warning(
+            f"The last {diff} runs were not submitted to the runner,"
+            f" as all of the available {PREFECT_RUNNER_PROCESS_LIMIT.value()}"
+            " slots were occupied. To increase the number of available slots,"
+            " configure the `PREFECT_RUNNER_PROCESS_LIMIT` setting."
+        )
+
+    return submitted_run_ids
+
+
+@sync_compatible
+async def wait_for_submitted_runs(
+    flow_run_ids: Optional[Set[uuid.UUID]] = None,
+    task_run_ids: Optional[Set[uuid.UUID]] = None,
+    timeout: Optional[float] = None,
+    poll_interval: float = 3.0,
+):
+    """
+    Wait for completion of any provided flow and task runs, as well as runs submitted to the
+    current runner webserver, within a timeout.
+
+    Args:
+        flow_run_ids: Additional flow run IDs to wait for.
+        task_run_ids: Additional task run IDs to wait for. # TODO: /task/run endpoint
+        timeout: How long to wait for completion of all runs (seconds).
+        poll_interval: How long to wait between polling each run's state (seconds).
+    """
+    flow_run_ids = flow_run_ids or set()
+    if task_run_ids:
+        raise NotImplementedError("Waiting for task runs is not yet supported.")
+
+    async def wait_for_final_state(
+        run_type: Literal["flow", "task"], run_id: uuid.UUID
+    ):
+        read_run_method = getattr(client, f"read_{run_type}_run")
```

Comment:
I think I'm lost in the diff + enclosing scope -- note to double check where this client is coming from.

Reply:
yeah it's kind of a weird thing. at the time of calling it's in scope but it's not the most readable. although we do do this elsewhere in server.py